src/main/sensors/battery.c

```diff
@@ -84,9 +83,9 @@ static void updateBatteryVoltage(void)
         vBatFilterIsInitialised = true;
     }
 
-    #ifdef USE_ESC_TELEMETRY
-    if (batteryConfig->batteryMeterType == BATTERY_SENSOR_ESC && isEscTelemetryActive()) {
-        vbatLatest = getEscTelemetryVbat();
+    #ifdef USE_ESC_SENSOR
+    if (feature(FEATURE_ESC_SENSOR) && batteryConfig->batteryMeterType == BATTERY_SENSOR_ESC) {
```

Comment:
The only thing that actually breaks the configurator is https://github.com/basdelfos/betaflight/blob/180f4c045e0efae7fd8886a1c08d1a7f7f49fd7d/src/main/fc/fc_msp.c#L1661 . And that can easily be fixed by adding a length check like https://github.com/basdelfos/betaflight/blob/180f4c045e0efae7fd8886a1c08d1a7f7f49fd7d/src/main/fc/fc_msp.c#L1683 .